polaris/polaris/management/commands/watch_transactions.py

```diff
@@ -187,11 +193,33 @@ def find_matching_payment_op(
 
         return matching_payment_op
 
-    @staticmethod
-    def _check_payment_op(operation: Operation, want_asset: Asset) -> bool:
+    @classmethod
+    def _check_payment_op(cls, generic_operation: Operation, want_asset: Asset) -> bool:
+        if generic_operation.TYPE not in [
+            OperationType.PAYMENT,
+            OperationType.PATH_PAYMENT_STRICT_RECEIVE,
+            OperationType.PATH_PAYMENT_STRICT_SEND,
+        ]:
+            return False
+        operation = cls._cast_operation(generic_operation)
+        asset = operation.dest_asset.alpha_num4 or operation.dest_asset.alpha_num12
         return (
-            operation.type_code() == Xdr.const.PAYMENT
-            and str(operation.destination) == want_asset.distribution_account
-            and str(operation.asset.code) == want_asset.code
-            and str(operation.asset.issuer) == want_asset.issuer
+            str(operation.destination) == want_asset.distribution_account
```

Comment:
@overcat same here, can I just cast `operation` to get the `G...`/`M...` address? It looks like it returns an object representation:
```
class MuxedAccount:
    ...
    def __str__(self):
        out = []
        out.append(f"type={self.type}")
        out.append(f"ed25519={self.ed25519}") if self.ed25519 is not None else None
        out.append(f"med25519={self.med25519}") if self.med25519 is not None else None
        return f"<MuxedAccount {[', '.join(out)]}>"

```